ouroboros-consensus/src/Data/SOP/Strict.hs

```diff
@@ -60,6 +62,19 @@ type instance AllZipN    NP c = AllZip c
 type instance Prod       NP   = NP
 type instance SListIN    NP   = SListI
 
+instance forall k xs . (Typeable (NP k xs), All (Compose ToCBOR k) xs) => ToCBOR (NP k xs) where
```

Comment:
I think all the existing definitions use `f` for the tyvar instead of `k`. Please follow suit.